react-native/android/src/main/java/io/realm/react/RealmReactModule.java

```diff
@@ -18,19 +18,23 @@
 import fi.iki.elonen.NanoHTTPD;
 
 public class RealmReactModule extends ReactContextBaseJavaModule {
+    private static final String DEBUG_SHARED_OBJECT = "realmreact-dbg";
+    private static final String RELEASE_SHARED_OBJECT = "realmreact";
     private static final int DEFAULT_PORT = 8082;
     private static boolean sentAnalytics = false;
 
     private AndroidWebServer webServer;
     private Handler handler = new Handler(Looper.getMainLooper());
 
-    static {
-        SoLoader.loadLibrary("realmreact");
-    }
-
-    public RealmReactModule(ReactApplicationContext reactContext) {
+    public RealmReactModule(ReactApplicationContext reactContext, boolean debug) {
         super(reactContext);
 
+        if (debug) {
+            SoLoader.loadLibrary(DEBUG_SHARED_OBJECT);
+        } else {
+            SoLoader.loadLibrary(RELEASE_SHARED_OBJECT);
+        }
```

Comment:
you can't use reflection, because you don't know the package name of the app yet.